WalletWasabi.Fluent/Models/UI/ApplicationSettings.cs

```diff
@@ -212,4 +216,14 @@ private void ApplyUiConfigChanges()
 		_uiConfig.HideOnClose = HideOnClose;
 		_uiConfig.PrivacyMode = PrivacyMode;
 	}
+
+	private bool GetIsOverridden()
+	{
+		if (_persistentConfig.Network != _config.Network)
+		{
+			return true;
+		}
+
+		return false;
+	}
```

Comment:
It worries me a bit that we will have to compare values one by one here to determine whether something is overridden. There are many settings and probably many more will be added in the future.

This is error prone, as someone in the future might add new settings and forget to add the comparison here.

I wonder if there's a way to improve/abstract this away?

Reply:
On start, if you compare PersistentConfig against Config, and you see any difference then you can make sure something has been overridden.

Probably in ctor?
```c#
IsOverridden = _startupConfig.AreDeepEqual(config);
```

Reply:
@soosr the thing is that `Config` and `PersistentConfig` are different classes and there's no guarantee their members will exactly match.

Maybe that is the root problem.

Reply:
Can we then create a new `Config` without any program argument, and compare it to the one that received program argument?
And check if the values are the same?

Reply:
@soosr yes that would make sense as the `Config` would only be proxying the `PersistentConfig` values in that case.

Reply:
I've implemented a solution that consists of setting all the properties in the constructor of `Config`, thus being able to easily calculate a new property `IsOverridden`, which can be read later. This solution is more maintainable, although it required to modify the class. Please, give your opinions.

Reply:
As we discussed on Slack I'm okay with it, but would be nicer if it was a separate PR and code team could review it